src/utils/module-cache.ts

```diff
@@ -23,18 +22,18 @@ export class ModuleCache {
     return result
   }
 
-  get<T>(cacheKey: CacheKey, settings: ImportSettings['cache']): T | undefined {
-    if (this.map.has(cacheKey)) {
-      const f = this.map.get(cacheKey)
+  get<T>(cacheKey: string, settings: NormalizedCacheSettings): T | undefined {
+    const cache = this.map.get(cacheKey)
+    if (cache) {
       // check freshness
-      // @ts-expect-error TS can't narrow properly from `has` and `get`
-      if (process.hrtime(f.lastSeen)[0] < settings.lifetime) {
-        return f!.result as T
+      if (process.hrtime(cache.lastSeen)[0] < settings.lifetime) {
+        return cache.result as T
       }
     } else {
       log('cache miss for', cacheKey)
     }
     // cache miss
+    clearNodeResolverCache()
```

Comment:
We may need to add an optional `clearResolveCache` method into v3 interface?

cc @43081j @SukkaW

Reply:
That's weird. TBH, this `ModuleCache` is only a TTL map that respects fs `mtime`. I am really not a fan of a double cache layer here.

What about adding the `cacheTtl` option to `unrs-resolver`, and we simply pass `settings.lifetime` to that? And then we can safely throw away this `ModuleCache`.

Or if we do want to retain the `ModuleCache`, we might just disable `unrs-resolver`'s cache completely.

Reply:
There could be 3rd-party resolvers, their own cache is not cleared previously, so without a new `clearResolveCache` method, 3rd-party's cache can never be cleared, for example `eslint-import-resolver-oxc`.

Reply:
As I mentioned above, 3rd-party resolvers like `eslint-import-resolver-oxc` has no chance to clear its own cache, it's not only about `unrs-resolver`, and according to https://github.com/jestjs/jest/issues/15660 and https://github.com/jestjs/jest/pull/15666, a singleton resolver instance with reused cache has much better performance.

Actually I want to do the opposite, the resolvers should respect the cache settings instead, we shouldn't cache for all resolvers which could be unexpected for 3rd-party resolvers, so my original proposal is adding an optional `clearResolveCache` method.

Reply:
> As I mentioned above, 3rd-party resolvers like `eslint-import-resolver-oxc` has no chance to clear its own cache

I have not caught up yet. Why do you want to clear the resolvers' cache on `eslint-plugin-import-x`'s internal `ModuleCache` miss? Does `eslint-plugin-import-x`'s internal cache have anything to do with downstream resolvers?

Reply:
I believe the PR could continue now.